src/control/positionControl/PositionControl.cpp

```diff
@@ -6,44 +6,215 @@
+            auto intermediateToTarget = calculatePathFromNewStart(intermediatePathCollision, pathToIntermediatePoint,
+                                                                  targetPosition, robotId, timeStep);
+            if (intermediateToTarget.has_value()) {
+                interface::Input::drawData(interface::Visual::PATHFINDING, intermediatePoints, Qt::green, robotId,
+                                           interface::Drawing::CROSSES);
+                interface::Input::drawData(interface::Visual::PATHFINDING, {firstCollision->collisionPosition},
+                                           Qt::red, robotId, interface::Drawing::CROSSES);
+
+                interface::Input::drawData(interface::Visual::PATHFINDING,
+                                           pathToIntermediatePoint.getPathApproach(timeStep),
+                                           Qt::white, robotId,
+                                           interface::Drawing::LINES_CONNECTED);
+                interface::Input::drawData(interface::Visual::PATHFINDING,
+                                           intermediateToTarget.value().getPathApproach(timeStep),
+                                           Qt::yellow, robotId, interface::Drawing::LINES_CONNECTED);
+                return intermediateToTarget.value();
+            }
+            intermediatePointsSorted.pop();
+        }
+        return std::nullopt;
     }
 
-    interface::Input::drawData(interface::Visual::PATHFINDING, computedPaths[robotId], Qt::green, robotId, interface::Drawing::LINES_CONNECTED);
-    interface::Input::drawData(interface::Visual::PATHFINDING, {computedPaths[robotId].front(), currentPosition}, Qt::green, robotId, interface::Drawing::LINES_CONNECTED);
-    interface::Input::drawData(interface::Visual::PATHFINDING, computedPaths[robotId], Qt::blue, robotId, interface::Drawing::DOTS);
+    std::vector<Vector2>
+    PositionControl::createIntermediatePoints(const rtt::world::Field &field, int robotId, std::optional<BB::CollisionData> &firstCollision,
+                                              Vector2 &targetPosition) {
+        double angleBetweenIntermediatePoints = M_PI_4 / 2;
 
-    RobotCommand command = RobotCommand();
-    command.pos = computedPaths[robotId].front();
-    Position trackingVelocity = pathTrackingAlgorithm.trackPathDefaultAngle(currentPosition, currentVelocity, computedPaths[robotId], robotId, pidType);
-    command.vel = Vector2(trackingVelocity.x, trackingVelocity.y);
-    command.angle = trackingVelocity.rot;
+        // PointToDrawFrom is picked by drawing a line from the target position to the obstacle and extending that
+        // line further towards our currentPosition.
+        Vector2 pointToDrawFrom = firstCollision->obstaclePosition +
+                                  (firstCollision->obstaclePosition - targetPosition).normalize() * 1;
```

Comment:
redundant *1?

Reply:
1 is the length of the extension. Made it more readable by adding an extension variable with value 1